test/Groups/Permutations.jl

```diff
@@ -37,13 +33,251 @@
   p[8] = cperm(G,[1,2,3,4,5,6,7],[8,9,10,11,12,13,14])
   p[9] = cperm(G,[1,2],[10,11])
   @test gens == p
-  
-  @test_throws ArgumentError @perm 10 [(1,11)]
-  
+    
   G = sub(G,gens)[1]
   @test order(G) == 645120
 end
 
+@testset "@perm" begin
+  @testset "@perm <...> cycles" begin
+    @testset "@perm cycles" begin
+      p = @perm (1,2,6)(4,5)
+      @test p isa PermGroupElem
+      G = parent(p)
+      @test degree(G) == 6
+      @test is_natural_symmetric_group(G)
+      @test order(p) == 6
+  
+      p = @perm ()
+      @test p isa PermGroupElem
+      G = parent(p)
+      @test degree(G) == 1
+      @test is_natural_symmetric_group(G)
+      @test order(p) == 1
+  
+      @test_throws ArgumentError @perm (-1,1)
+    end
+
+    @testset "@perm n cycles" begin
+      n = 7
+
+      p = @perm n (1,2,6)(4,5)
+      @test p isa PermGroupElem
+      G = parent(p)
+      @test degree(G) == n
+      @test is_natural_symmetric_group(G)
+      @test order(p) == 6
+
+      p = @perm n ()
+      @test p isa PermGroupElem
+      G = parent(p)
+      @test degree(G) == n
+      @test is_natural_symmetric_group(G)
+      @test order(p) == 1
+
+      @test_throws ArgumentError @perm 10 (1,11)
+      @test_throws ArgumentError @perm 5 (-1,1)
+    end
+
+    @testset "@perm G cycles" begin
+      S = symmetric_group(7)
+      p = @perm S (1,2,6)(4,5)
+      @test p isa PermGroupElem
+      @test parent(p) == S
+      @test order(p) == 6
+      p = @perm S ()
+      @test p isa PermGroupElem
+      @test parent(p) == S
+      @test order(p) == 1
+
+      @test_throws ArgumentError @perm S (-1,1)
+      @test_throws ArgumentError @perm S (2,5,8)
+
+      A = alternating_group(7)
+      p = @perm A (1,6)(4,5)
+      @test p isa PermGroupElem
+      @test parent(p) == A
+      @test order(p) == 2
+      p = @perm A ()
+      @test p isa PermGroupElem
+      @test parent(p) == A
+      @test order(p) == 1
+
+      # perms with negative sign
+      @test_throws ArgumentError @perm A (3,5) 
+      @test_throws ArgumentError @perm A (1,2,3,4) 
+      @test_throws ArgumentError @perm A (2,5)(4,1,3) 
+    end
+  end
+
+  @testset "@perm <...> vector" begin
+    let n = 7
+      S = symmetric_group(n)
+      @static if VERSION >= v"1.7"
+        # the following tests need the improved `@macroexpand` from Julia 1.7
+        @test_throws ArgumentError @macroexpand @perm []
+        @test_throws ArgumentError @macroexpand @perm n []
+        @test_throws ArgumentError @macroexpand @perm S []
+      end
+    end
+
+    @testset "@perm vector" begin
+      ps = @perm [(1,4)(2,3)(6,5)]
+      @test ps isa Vector{PermGroupElem}
+      G = parent(ps[1])
+      @test degree(G) == 6
+      @test is_natural_symmetric_group(G)
+      @test ps == [@perm(G, (1,4)(2,3)(6,5))]
+      
+      ps = @perm [(1,2,3)(4,5), (6,2,3,1), (), (2,)(1,4)]
+      @test ps isa Vector{PermGroupElem}
+      @test allequal(parent, ps)
```

Comment:
`allequal` requires Julia >= 1.8 and we still run some of these tests in Julia 1.6.

Perhaps we should provide an `allequal` implementation for older Julia versions?

Reply:
Isn't this what `Compat.jl` is for? We already have it in the environment as a dependency of GAP.jl anyway. For the time being, I just import this one function in older julia versions.

(I don't understand why the 1.10 job succeeded, as the `allequal` dispatch with predicate was only added right before the 1.11 feature freeze)

Reply:
Thanks for reminding me of https://github.com/JuliaLang/Compat.jl/issues/814. This indeed resolves my confusion.